test_runner/src/main/kotlin/ftl/args/ValidateAndroidArgs.kt

```diff
@@ -171,17 +172,23 @@ private fun AndroidArgs.assertMaxTestShardsByDeviceType() =
     }
 
 private fun AndroidArgs.assertDevicesShards() {
-    if (inVirtualRange && !inPhysicalRange) logLn("Physical devices configured, but max-test-shards limit set to $maxTestShards, for physical devices range is ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.first} to ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.last}, you additionally have configured virtual devices. In this case, the physical limit will be decreased to: ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.last}")
-    else if (!inVirtualRange && !inPhysicalRange) throwMaxTestShardsLimitExceeded()
+    when {
+        inVirtualRange && !inPhysicalRange -> logLn("Physical devices configured, but max-test-shards limit set to $maxTestShards. For physical devices range is ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.first} to ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.last}, you additionally have configured virtual devices. In this case, the maximum shards will be decreased to: ${IArgs.AVAILABLE_PHYSICAL_SHARD_COUNT_RANGE.last}")
+        !inVirtualRange && !inPhysicalRange -> throwMaxTestShardsLimitExceeded()
+    }
 }
 
 private fun AndroidArgs.assertVirtualDevicesShards() {
-    if (!inVirtualRange) throwMaxTestShardsLimitExceeded()
+    when {
+        devices.containsARMDevices() && !devices.containsARMDevices() -> logLn("ARM devices configured, but max-test-shards limit set to $maxTestShards. For ARM devices, the range is ${IArgs.AVAILABLE_VIRTUAL_ARM_SHARD_COUNT_RANGE.first} to ${IArgs.AVAILABLE_VIRTUAL_ARM_SHARD_COUNT_RANGE.last}, you additionally have configured other virtual devices. In this case, the maximum shards will be decreased to: ${IArgs.AVAILABLE_VIRTUAL_ARM_SHARD_COUNT_RANGE.last}")
```

Comment:
Is this the right conditional? This appears to always be false?

Reply:
updated